### ISSUE TITLE
Issue with rail sim landing and some other bugs

### ISSUE THREAD

**Issue by BAmercury** [BAmercury/ardupilot_gazebo#36]
Looks like at the end of the new motion profile 4 there is a bug (In gazebo 9 at least, not sure about gazebo 7) where the drone will land but not disarm when it touches the platform, it'll just sort of spaz out. This does not happen in the iris_irlock_demo.world so it's specific to the rail sim one. Anyways, I think it might be that at the end of the motion profile 4 it'll spam the rail sim to stay at the origin. Adding some more control logic so that command gets issues only once and maybe that'll fix it. 

Also had some trouble actually getting Ardupilot SITL to talk to the iris_irlock_demo.world, added some changes in and now that seems to be working - not sure exactly what was going with it but it would time out after trying to configure the gyro. I added a pose tag for the iris drone and now it spawns on top of the landing platform and added some physics changes for the rail platform (update the inertia,etc. ) and now it seems to be communicating again. Need to run some more tests but maybe it would be benefical to have the iris_irlock_rail_sim.world have a motion profile where it just stands still and eliminates the need for the iris_irlock_demo.world file. So it's all in one place. 

These changes are all on Gazebo 9 master and then will port over to Gazebo 7 master

**Comment by BAmercury** [comment]
Okay adding that extra control boolean seemed to do the trick. I guess because the plugin was continuously telling the Rail sim to stay at a specfic position something got messed up with the contact physics so the drone couldn't land properly. Going to go back and look into the static target world and make sure that is working correctly before pushing changes to gazebo9 master

**Comment by BAmercury** [comment]
Yeah so for the static world (iris_irlock_demo.world) all I really added in was:

- high inertia for the landing pad

- pose tag for the iris_drone


Works now but will probably pilot another issue if this comes up again. Final addition I'll go ahead and have the rail sim and drone spawn at the origin

**Comment by BAmercury** [comment]
Alright adding these changes into Gazebo 7 then closing this issue out. Will reopen more bug reports if needed
